src/app/core/services/basic-rundown-state.service.ts

```diff
@@ -11,28 +11,28 @@ import { RundownEvent } from '../models/rundown-event'
 export class BasicRundownStateService implements OnDestroy {
   private readonly basicRundownsSubject: BehaviorSubject<BasicRundown[]>
   private basicRundowns: BasicRundown[] = []
-  private unsubscribeFromEvents?: Unsubscribe
+  private eventSubscriptions: EventSubscription[]
 
   constructor(
       private readonly basicRundownService: BasicRundownService,
       private readonly rundownEventObserver: RundownEventObserver,
       private readonly connectionStatusObserver: ConnectionStatusObserver
   ) {
     this.basicRundownsSubject = new BehaviorSubject<BasicRundown[]>(this.basicRundowns)
-    this.registerEventConsumers()
+    this.subscribeToEvents()
     this.resetBasicRundownSubject()
   }
 
-  private registerEventConsumers(): void {
-    const unsubscribeFromConnectionStatusEvents = this.registerConnectionStatusConsumers()
-    const unsubscribesFromRundownEvents = this.registerBasicRundownEventConsumers()
-    this.unsubscribeFromEvents = () => [
-        ...unsubscribesFromRundownEvents,
-        ...unsubscribeFromConnectionStatusEvents
-      ].forEach(unsubscribe => unsubscribe())
+  private subscribeToEvents(): void {
+    const connectionStatusSubscriptions = this.subscribeToConnectionStatus()
```

Comment:
Should these not be singular? Don't you just subscribe a single time to a pipeline of events? The name indicates that the constant holds multiple subscriptions.  Same question for the one underneath, and `subscribeToEvents()` in the `rundown-state.service.ts`.

Reply:
You are correct that currently `subscribeToConnectionStatus` returns a list of subscriptions with a length of 1. `subscribeToRundownEvents` returns a list with more than one subscription – one for each type of event that it subscribes to. One could wrap this in an EventSubscription object, to have a single object to unsubscribe from all subscriptions with a "single" call.